src/csvcubed/writers/qbwriter.py

```diff
@@ -540,19 +540,19 @@ def _get_primary_key_columns(self) -> List[str]:
         return primary_key_columns
 
     def _determine_whether_column_is_required(self, column: QbColumn) -> bool:
-        return (
-            isinstance(
-                column.structural_definition,
-                (QbDimension, QbMultiUnits, QbMultiMeasureDimension),
-            )
-            or (
-                isinstance(column.structural_definition, QbAttribute)
-                and column.structural_definition.get_is_required()
-            )
-            or (
-                isinstance(column.structural_definition, QbObservationValue)
-                and len(get_observation_status_columns(self.cube)) == 0
-                # We cannot mark an observation value column as `required` if there are `obsStatus` columns defined
-                # since we permit missing observation values where an `obsStatus` explains the reason.
-            )
-        )
+        if isinstance(column.structural_definition,(QbDimension, QbMultiUnits, QbMultiMeasureDimension)):
+            return True
+        elif (isinstance(column.structural_definition, QbAttribute) and column.structural_definition.get_is_required()):
+            if self.cube.is_pivoted_shape:
```

Comment:
It should be possible to have a required attribute in a single measure pivoted data set. We need to check whether it's pivoted **and** it's a multi-obs-val data set in this condtion here.